thavalon-server/src/game/state/on_mission.rs

```diff
@@ -244,8 +244,17 @@ fn conclude_mission<P: Phase>(
         log::debug!("3 missions have failed, the Evil team has won");
         state.into_done(Team::Evil, effects)
     } else {
-        let mission_proposer = &state.proposals[proposal].proposer;
-        let next_proposer = state.game.next_proposer(mission_proposer).to_string();
+        let mission = state.mission();
+        let next_proposer = if mission == 2 {
+            // On mission 2, the third proposer goes first, because the first 2 proposers already proposed for mission 1
+            // the mod is to account for 2-player testing games, for which we wrap back around to player 1
+            let next_proposer_index = 3 % state.game.proposal_order.len();
```

Comment:
There is an off by one error here I think with mod 3, at least in five player games, leading the third proposer to be skipped.

Reply:
Oh, it's because the 3rd proposer is at index 2, not index 3 😬